dev/tasks/tasks.yml

```diff
@@ -15,81 +15,224 @@
     artifacts:
       - arrow-cpp-{version}-py27_0-osx-64.tar.bz2
-      - arrow-cpp-{version}-py35_0-osx-64.tar.bz2
-      - arrow-cpp-{version}-py36_0-osx-64.tar.bz2
       - pyarrow-{version}-py27_0-osx-64.tar.bz2
-      - pyarrow-{version}-py35_0-osx-64.tar.bz2
-      - pyarrow-{version}-py36_0-osx-64.tar.bz2
       - parquet-cpp-1.4.0-0-osx-64.tar.bz2
 
-  conda-win:
+  conda-osx-py35:
+    platform: osx
+    template: conda-recipes/travis.osx.yml
+    params:
+      variant_config_file: variants/osx_python3.5.yaml
+    artifacts:
+      - arrow-cpp-{version}-py35_0.tar.bz2
+      - pyarrow-{version}-py35_0.tar.bz2
+      - parquet-cpp-1.4.0-0.tar.bz2
+
+  conda-osx-py36:
+    platform: osx
+    template: conda-recipes/travis.osx.yml
+    params:
+      variant_config_file: variants/osx_python3.6.yaml
+    artifacts:
+      - arrow-cpp-{version}-py36_0.tar.bz2
+      - pyarrow-{version}-py36_0.tar.bz2
+      - parquet-cpp-1.4.0-0.tar.bz2
+
+  ############################## Conda Windows ################################
+
+  conda-win-py35:
     platform: win
     template: conda-recipes/appveyor.yml
+    params:
+      conda_install_location: C:\\Miniconda35-x64
+      variant_config_file: variants\win_c_compilervs2015cxx_compilervs2015python3.5.yaml
     artifacts:
       - arrow-cpp-{version}-py35_vc14_0.tar.bz2
-      - arrow-cpp-{version}-py36_vc14_0.tar.bz2
       - pyarrow-{version}-py35_vc14_0.tar.bz2
+      - parquet-cpp-1.4.0-vc14_0.tar.bz2
+
+  conda-win-py36:
+    platform: win
+    template: conda-recipes/appveyor.yml
+    params:
+      conda_install_location: C:\\Miniconda36-x64
+      variant_config_file: variants\win_c_compilervs2015cxx_compilervs2015python3.6.yaml
+    artifacts:
+      - arrow-cpp-{version}-py36_vc14_0.tar.bz2
       - pyarrow-{version}-py36_vc14_0.tar.bz2
       - parquet-cpp-1.4.0-vc14_0.tar.bz2
 
-  # python wheels
-  wheel-linux:
+  ############################## Wheel Linux ##################################
+
+  wheel-linux-cp27m:
     platform: linux
     template: python-wheels/travis.linux.yml
+    params:
+      python_version: 2.7,16
     artifacts:
       - pyarrow-{version}-cp27-cp27m-manylinux1_x86_64.whl
+
+  wheel-linux-cp27mu:
+    platform: linux
+    template: python-wheels/travis.linux.yml
+    params:
+      python_version: 2.7,32
+    artifacts:
       - pyarrow-{version}-cp27-cp27mu-manylinux1_x86_64.whl
+
+  wheel-linux-cp35m:
+    platform: linux
+    template: python-wheels/travis.linux.yml
+    params:
+      python_version: 3.5,16
+    artifacts:
       - pyarrow-{version}-cp35-cp35m-manylinux1_x86_64.whl
+
+  wheel-linux-cp36m:
+    platform: linux
+    template: python-wheels/travis.linux.yml
+    params:
+      python_version: 3.6,16
+    artifacts:
       - pyarrow-{version}-cp36-cp36m-manylinux1_x86_64.whl
 
-  wheel-osx:
+  ############################## Wheel OSX ####################################
+
+  wheel-osx-cp27m:
```

Comment:
I know expanding to multiple tasks is a bit bulky, but it's explicit at least. Later we can implement a parameter matrix, but We wouldn't spare too many LoCs (but would introduce implicit things like generating task names).